cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+		// Later on we don't use the app name retrieved here since we can use it from the manifest.
+		// Here we simply fail early when the app name is not provided and also not contained in the manifest.
+		appName, err = getAppNameOrFail(config)
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Entry().Infof("CF native deployment ('%s') with:", config.DeployType)
+	log.Entry().Infof("cfAppName='%s'", appName)
+	log.Entry().Infof("cfManifest='%s'", config.Manifest)
+	log.Entry().Infof("cfManifestVariables: '%v'", config.ManifestVariables)
+	log.Entry().Infof("cfManifestVariablesFiles: '%v'", config.ManifestVariablesFiles)
+	log.Entry().Infof("cfdeployDockerImage: '%s'", config.DeployDockerImage)
+	log.Entry().Infof("smokeTestScript: '%s'", config.SmokeTestScript)
+
+	additionalEnvironment := []string{
+		"STATUS_CODE=" + strconv.FormatInt(int64(config.SmokeTestStatusCode), 10),
+	}
+
+	if len(config.DockerPassword) > 0 {
+		additionalEnvironment = append(additionalEnvironment, "CF_DOCKER_PASSWORD="+config.DockerPassword)
+	}
+
+	myDeployConfig := deployConfig{
+		DeployCommand:   deployCommand,
+		DeployOptions:   deployOptions,
+		AppName:         config.AppName,
+		ManifestFile:    config.Manifest,
+		SmokeTestScript: smokeTestScript,
+	}
+
+	log.Entry().Infof("DeployConfig: %v", myDeployConfig)
+
+	//return nil
+	return deployCfNative(myDeployConfig, config, additionalEnvironment, command)
+}
+
+func deployCfNative(deployConfig deployConfig, config *cloudFoundryDeployOptions, additionalEnvironment []string, cmd command.ExecRunner) error {
+
+	deployStatement := []string{
+		deployConfig.DeployCommand,
+	}
+
+	if len(deployConfig.AppName) > 0 {
+		deployStatement = append(deployStatement, deployConfig.AppName)
+	}
+
+	if len(deployConfig.DeployOptions) > 0 {
+		deployStatement = append(deployStatement, deployConfig.DeployOptions...)
+	}
+
+	if len(deployConfig.ManifestFile) > 0 {
+		deployStatement = append(deployStatement, "-f")
+		deployStatement = append(deployStatement, deployConfig.ManifestFile)
+	}
+
+	if len(config.DeployDockerImage) > 0 && config.DeployType != "blue-green" {
+		deployStatement = append(deployStatement, "--docker-image", config.DeployDockerImage)
+	}
+
+	if len(config.DockerUsername) > 0 && config.DeployType != "blue-green" {
+		deployStatement = append(deployStatement, "--docker-username", config.DockerUsername)
+	}
+
+	if len(deployConfig.SmokeTestScript) > 0 {
+		deployStatement = append(deployStatement, deployConfig.SmokeTestScript...)
+	}
+
+	if len(config.CfNativeDeployParameters) > 0 {
+		deployStatement = append(deployStatement, strings.FieldsFunc(config.CfNativeDeployParameters, func(c rune) bool {
+			return c == ' '
+		})...)
+	}
+
+	stopOldAppIfRunning := func(_cmd command.ExecRunner) error {
+
+		if config.KeepOldInstance && config.DeployType == "blue-green" {
+			oldAppName := deployConfig.AppName + "-old"
+			_ = oldAppName
+
+			var buff bytes.Buffer
+
+			oldOut := _cmd.Stdout
+			_ = oldOut
```

Comment:
aaaahhh, that is some kind of a left over. Basically I would like to remember the writer which is currently set on the `_cmd` in order to be able to restore it afterwards. Some lines below I simply set `_cmd.Stdout(log.Writer())` again which might end up in a correct result, but is in fact not correct. The correct way is to remember what we have earlier and to reset that instead of simply using log.Writer(). I will provide another PR for getting the currently set writer on `_cmd` so that we can improve that later. I already opened a PR for that, but closed it since I thought there wouldn't be any need for taht (was PR #1787).

Reply:
Stupid lines of codes has been removed.